src/neynarAPI/NeynarAPIClient.ts

```diff
@@ -0,0 +1,679 @@
+    if (!(error instanceof AxiosError)) return false;
+    return error.response?.data !== undefined;
+  }
+
+  /**
+   * Creates a Signer. See [Neynar documentation](https://docs.neynar.com/reference/create-signer)
+   * for more details.
+   */
+  public async createSigner(): Promise<Signer> {
+    const response = await this.apis.signer.createSigner();
+    return response.data;
+  }
+
+  /**
+   * Fetches an existing Signer. See [Neynar documentation](https://docs.neynar.com/reference/get-signer)
+   * for more details.
+   *
+   */
+  public async fetchSigner(signerUuid: string): Promise<Signer | undefined> {
+    try {
+      const response = await this.apis.signer.signer(signerUuid);
+      return response.data;
+    } catch (error) {
+      if (NeynarAPIClient.isApiErrorResponse(error)) {
+        const status = error.response.status;
+        if (status === 404) {
+          return undefined;
+        }
+      }
+      throw error;
+    }
+  }
+
+  /**
+   * Registers a Signer with an fid. See [Neynar documentation](https://docs.neynar.com/reference/register-app-fid)
+   * for more details.
+   */
+  public async registerSigner(
+    signerUuid: string,
+    fid: number,
+    deadline: number,
+    signature: string
+  ): Promise<Signer> {
+    const body: RegisterSignerKeyReqBody = {
+      signer_uuid: signerUuid,
+      app_fid: fid,
+      deadline,
+      signature,
+    };
+    const response = await this.apis.signer.registerSignedKey(body);
+    return response.data;
+  }
+
+  /**
+   * Gets information about an individual cast using neynar v1 API
+   */
+  public async v1FetchCast(
+    castOrCastHash: v1Cast | string
+  ): Promise<v1Cast | undefined> {
+    let castHash: string;
+    if (typeof castOrCastHash === "string") {
+      castHash = castOrCastHash;
+    } else {
+      castHash = castOrCastHash.hash;
+    }
+    const response = await this.apis.v1Cast.cast(castHash);
+    if (response.status === 404) return undefined;
+    return response.data.result.cast ?? undefined;
+  }
+
+  /**
+   * Gets information about an individual cast
+   */
+  public async fetchCast(
+    castOrCastHash: v2Cast | string
+  ): Promise<v2Cast | undefined> {
+    let castHash: string;
+    if (typeof castOrCastHash === "string") {
+      castHash = castOrCastHash;
+    } else {
+      castHash = castOrCastHash.hash;
+    }
+    try {
+      const response = await this.apis.v2Cast.cast(
+        CastParamType.Hash,
+        castHash
+      );
+      return response.data.cast ?? undefined;
+    } catch (error) {
+      if (NeynarAPIClient.isApiErrorResponse(error)) {
+        const status = error.response.status;
+        if (status === 404) {
+          return undefined;
+        }
+      }
+      throw error;
+    }
+  }
+
+  /**
+   * Publishes a cast for the currently authenticated user
+   */
+  public async publishCast(
+    signerUuid: string,
+    text: string,
+    replyTo?: v2Cast | { hash: string },
```

Comment:
How do we handle FIP-2 (reply to arbitrary strings)?
https://github.com/farcasterxyz/protocol/discussions/71

Reply:
I'll update the method signature to accept strings and add a test.